src/parser/paladin/protection/modules/talents/Seraphim.js

```diff
@@ -15,18 +16,34 @@ const SERAPHIM_STAT_BUFF = 1007;
 class Seraphim extends Analyzer {
   static dependencies = {
     spellUsable: SpellUsable,
+    abilities: Abilities,
   };
 
+  lastCDConsumed = null;
+
   constructor(...args) {
     super(...args);
     this.active = this.selectedCombatant.hasTalent(SPELLS.SERAPHIM_TALENT.id);
   }
 
+  sotrCooldown() {
+    if(this.spellUsable.isOnCooldown(SPELLS.SHIELD_OF_THE_RIGHTEOUS.id)) {
+      return this.spellUsable.cooldownRemaining(SPELLS.SHIELD_OF_THE_RIGHTEOUS.id);
+    } else {
+      return 0;
+    }
+  }
+
   on_byPlayer_cast(event) {
     if (event.ability.guid !== SPELLS.SERAPHIM_TALENT.id) {
       return;
     }
 
+    const expectedCd = this.abilities.getExpectedCooldownDuration(SPELLS.SHIELD_OF_THE_RIGHTEOUS.id, event);
+    const chargesAtCast = 3 - this.sotrCooldown() / expectedCd;
+    this.lastCDConsumed = expectedCd * Math.min(2, chargesAtCast);
+
+    console.log(`Charges at cast ${chargesAtCast}`);
```

Comment:
Any reason this isn't surrounded by a if(debug)? Feels like we are adding bloat to console